Number of points requested cannot exceed 200

alliejanoch opened the pull request:
NOAA will only process weather requests for 200 points at a time. This breaks the list of lat/lon's into batches of 200. 

(http://graphical.weather.gov/xml/ about 1/2 way down there is a note about this)

@thataustin - if you want, I can take over this library. Up to you.

alliejanoch committed:
Add some debugging

alliejanoch committed:
request forecast in chunks of no more than 200

alliejanoch committed:
gracefully handle sites outside of the usa

gracefully handle sites outside of the usa

thataustin commented on index.js:
```diff
@@ -31,18 +31,16 @@ var noaaForecaster = {
-        var options = _.extend({}, forecastDetails, { listLatLon: listLatLon});
-        var url = this._convertOptionsToUrlString('data', options);
-        return this._makeCall(url);
+        return this._getForecastPerPoint(geoBusinessObjects, forecastDetails, chunkSize);
```
I think the chunkSize could be an internal detail to the `getForecastPerPoint` function.  That function should fetch in chunks and return a flattened version of the results.  The next function in the chain doesn't have to know or care that there's a chunk limitation in the `_getForecastPerPoint` process.

thataustin commented:
Sorry, I just saw this!  Holy cow, time to quit using the "Mark all as read" button in Github alerts.

alliejanoch committed:
Allow lat,lng pairs outside the us to be removed if first in batch

print url when there is an error

handle bad lat lng being first in batch of 200

handle bad lat lng being first in batch of 200

handle bad lat lng being first in batch of 200

handle bad lat lng being first in batch of 200

handle bad lat lng being first in batch of 200

alliejanoch committed:
move chunksize to just be in one func

alliejanoch committed:
A little code clean up

alliejanoch commented:
@thataustin - good idea with moving chunk size. I did that but I've also added some other stuff since this pull request was originally created: locations outside the NOAA's coverage were causing errors, so I now remove them from the list of lat/lng pairs being processed.
